lntest/itest/lnd_channel_graph_test.go

```diff
@@ -370,47 +370,33 @@ func testGraphTopologyNtfns(ht *lntemp.HarnessTest, pinned bool) {
 // testNodeAnnouncement ensures that when a node is started with one or more
 // external IP addresses specified on the command line, that those addresses
 // announced to the network and reported in the network graph.
-func testNodeAnnouncement(net *lntest.NetworkHarness, t *harnessTest) {
-	ctxb := context.Background()
-
-	aliceSub := subscribeGraphNotifications(ctxb, t, net.Alice)
-	defer close(aliceSub.quit)
+func testNodeAnnouncement(ht *lntemp.HarnessTest) {
+	alice, bob := ht.Alice, ht.Bob
 
 	advertisedAddrs := []string{
 		"192.168.1.1:8333",
 		"[2001:db8:85a3:8d3:1319:8a2e:370:7348]:8337",
 		"bkb6azqggsaiskzi.onion:9735",
-		"fomvuglh6h6vcag73xo5t5gv56ombih3zr2xvplkpbfd7wrog4swjwid.onion:1234",
+		"fomvuglh6h6vcag73xo5t5gv56ombih3zr2xvplkpbfd7wrog4swj" +
+			"wid.onion:1234",
 	}
 
 	var lndArgs []string
 	for _, addr := range advertisedAddrs {
 		lndArgs = append(lndArgs, "--externalip="+addr)
 	}
 
-	dave := net.NewNode(t.t, "Dave", lndArgs)
-	defer shutdownAndAssert(net, t, dave)
+	dave := ht.NewNode("Dave", lndArgs)
 
 	// We must let Dave have an open channel before he can send a node
 	// announcement, so we open a channel with Bob,
-	net.ConnectNodes(t.t, net.Bob, dave)
-
-	// Alice shouldn't receive any new updates yet since the channel has yet
-	// to be opened.
-	select {
-	case <-aliceSub.updateChan:
```

Comment:
why remove this one? is it perhaps getting triggered by other announcements and causing this to flake?

Reply:
this is prob something I want to improve in the future to make the test more self-explanatory. So this is now removed because we handle all our topology changes inside [node watcher](https://github.com/lightningnetwork/lnd/blob/master/lntemp/node/watcher.go#L286), and the method `subscribeGraphNotifications` is removed. Previously we did see flakes coming from the fact that we made a subscription too late, which missed events. And now the node watcher is started as soon as a new node is created so we never miss events.